api/models/Post.js

```diff
@@ -201,6 +203,44 @@ module.exports = bookshelf.Model.extend({
     })
   },
 
+  sendPushNotifications: function(opts) {
+    var uniqById = function(array) {
+      //this is destructive of array
+      var result = []
+      while(array.length > 0) {
+        var element = array[0]
+        result.push(element)
+        _.remove(array, item => item.get("id") == element.get("id"))
+      }
+      return result;
+    }
+
+    return Post.find(opts.postId, {withRelated: ['communities', 'communities.users', 'communities.users.communities', 'creator']})
+      .then((post) => {
+        var communities = post.relations.communities,
+          creator = post.relations.creator,
+          usersWithDupes = communities.map((community => community.relations.users.models)),
+          users = uniqById(_.flatten(usersWithDupes))
+
+        _.remove(users, user => user.get("id") == creator.get("id"))
+        return Promise.map(users, (user) => {
+          if (!user.get("push_new_post_preference")) return
+
+          var userCommunities = user.relations.communities.models,
+            postCommunities = communities.models,
+            communityIntersection, community, path, alertText
+
+          communityIntersection = _.remove(userCommunities, community => _.find(postCommunities, {'id': community.id}))
```

Comment:
_.intersection has the same problem as uniq: it uses SameValueZero for equality, which is not always true even if the models have the same id